src/amo/components/MastHead.js

```diff
@@ -35,15 +30,12 @@ export class MastHeadBase extends React.Component {
         </div>
         {children}
         <header className="MastHead-header">
-          {isHomePage
-            ? <h1 ref={(ref) => { this.title = ref; }}
-                className="MastHead-title MastHead-homepage">
-              {headerTitle}
-            </h1>
-            : <Link ref={(ref) => { this.title = ref; }} to="/"
-                className="MastHead-title">
-              {headerTitle}
-            </Link>}
+          <h1>
```

Comment:
FWIW: Since the markup needs to be conditional I originally left the link out when isHomepage is true since it's a link to the page you're on so it seemed superfluous.